routing/single_mwm_router.cpp

```diff
@@ -92,8 +92,8 @@ IRouter::ResultCode SingleMwmRouter::DoCalculateRoute(MwmSet::MwmId const & mwmI
   if (!FindClosestEdge(mwmId, finalPoint, finishEdge))
     return IRouter::EndPointNotFound;
 
-  RoadPoint const start(startEdge.GetFeatureId().m_index, startEdge.GetSegId());
-  RoadPoint const finish(finishEdge.GetFeatureId().m_index, finishEdge.GetSegId());
+  FakeVertex const start(startEdge.GetFeatureId().m_index, startEdge.GetSegId(), startPoint);
```

Comment:
Are you sure? What if startPoint is close to the end of the startEdge, but in the index graph starter it will be transformed to a backward segment.